yb-voyager/cmd/importSchema.go

```diff
@@ -86,24 +86,42 @@ func importSchema() error {
 
 	tconf.Schema = strings.ToLower(tconf.Schema)
 
+	tconfSchema := tconf.Schema
+	// setting the tconf schema to public here for initalisation to handle cases where non-public target schema
+	// is not created as it will be created with `createTargetSchemas` func, so not a problem in using public as it will be
+	// available always and this is just for initialisation of tdb and marking it nil again back.
+	tconf.Schema = "public"
+	tdb = tgtdb.NewTargetDB(&tconf)
+	err = tdb.Init()
+	if err != nil {
+		utils.ErrExit("Failed to initialize the target DB: %s", err)
+	}
+
 	if callhome.SendDiagnostics || getControlPlaneType() == YUGABYTED {
-		tconfSchema := tconf.Schema
-		// setting the tconf schema to public here for initalisation to handle cases where non-public target schema
-		// is not created as it will be created with `createTargetSchemas` func, so not a problem in using public as it will be
-		// available always and this is just for initialisation of tdb and marking it nil again back.
-		tconf.Schema = "public"
-		tdb = tgtdb.NewTargetDB(&tconf)
-		err := tdb.Init()
+		targetDBDetails = tdb.GetCallhomeTargetDBInfo()
+	}
+
+	if tconf.RunGuardrailsChecks {
+		// Check import schema permissions
+		missingPermissions, err := tdb.GetMissingImportSchemaPermissions()
 		if err != nil {
-			utils.ErrExit("Failed to initialize the target DB: %s", err)
+			utils.ErrExit("Failed to get missing import schema permissions: %s", err)
+		}
+		if len(missingPermissions) > 0 {
+			utils.PrintAndLog(color.RedString("The target database is missing the following permissions required for importing schema:"))
+			output := strings.Join(missingPermissions, "\n")
+			utils.PrintAndLog(output)
+			utils.ErrExit("Please grant the required permissions and retry the import.")
+		} else {
+			log.Info("The target database has the required permissions for importing schema.")
 		}
-		targetDBDetails = tdb.GetCallhomeTargetDBInfo()
-		//Marking tdb as nil back to not allow others to use it as this is just dummy initialisation of tdb 
-		//with public schema so Reintialise tdb if required with proper configs when it is available.
-		tdb = nil
-		tconf.Schema = tconfSchema
 	}
 
+	//Marking tdb as nil back to not allow others to use it as this is just dummy initialisation of tdb
+	//with public schema so Reintialise tdb if required with proper configs when it is available.
+	tdb = nil
+	tconf.Schema = tconfSchema
```

Comment:
will setting it to nil drops the connection?
Please check shall we Close() it separately.